frame/staking-rewards/src/lib.rs

```diff
@@ -378,13 +423,29 @@ pub mod pallet {
 		}
 
 		/// Extends fNFT position stake. Applied only to next epoch.
+		///
+		/// Is better then unstake and restake more in case of penalty.
+		/// But not incentivises to making many small stakes one on other to game over system.
 		#[pallet::weight(10_000)]
+		#[transactional]
 		pub fn extend_stake(
-			_origin: OriginFor<T>,
-			_instance_id: InstanceIdOf<T>,
-			_balance: T::Balance,
+			origin: OriginFor<T>,
+			instance_id: InstanceIdOf<T>,
+			balance: T::Balance,
+			keep_alive: bool,
 		) -> DispatchResult {
-			Err(DispatchError::Other("no implemented. TODO: insert update for next fold"))
+			// TODO: consider attack adding near zero stake each block, preventing stream to finish
```

Comment:
Shouldn't we set a minimum stake amount to call this then?

Reply:
there is separate task for that, just added additional TODO.

Reply:
there is task to handle that in clickup, just articulated concrere place